plugins/form/login/login.component.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path="./login.globals.ts"/>
+/// <reference path="../form.globals.ts"/>
+
+namespace FormAuthLogin {
+
+  const TOKEN_KEY = FormAuth.LOCAL_STORAGE_KEY_TOKEN;
+
+  export class LoginController {
+    branding: Core.Branding = { appName: '', appLogoUrl: '' };
+    login: Core.Login = {};
+    token: string;
+    invalidToken: boolean = false;
+    errorMessage: string;
+
+    constructor(
+      private configManager: Core.ConfigManager,
+      private $http: ng.IHttpService,
+      private $window: ng.IWindowService,
+      private localStorage: Storage,
+      private documentBase: string) {
+      'ngInject';
+    }
+
+    $onInit(): void {
+      this.branding = this.configManager.branding;
+      this.login = this.configManager.login;
+    }
+
+    doLogin(): void {
+      if (!this.token || this.token.trim() === '') {
+        this.invalidToken = true;
+        this.errorMessage = "Token is empty";
+        return;
+      }
+
+      this.validateToken(() => {
+        this.saveTokenAndRedirect();
+      });
+    }
+
+    private validateToken(callback: () => void): void {
+      log.debug("Fetching osconsole/config.js");
+      $.getScript('osconsole/config.js')
+        .always(() => {
+          const masterUri: string = _.get(window, 'HAWTIO_OAUTH_CONFIG.master_uri');
+          if (!masterUri) {
+            log.error('Master URI is not found');
+            this.invalidToken = true;
+            this.errorMessage = "Cannot validate token";
+            return;
+          }
+
+          this.$http
+            .get(masterUri, {
```

Comment:
Just a late comment, while doing some testing, I realise some service accounts may not be granted permission to access the root path of the API server, while there are generally granted permission to access the `api` path. This somehow relates to https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#without-kubectl-proxy.

Reply:
Thanks for testing, good catch!  I'll create another PR to fix it by using the master `/api` path.